tests/memory.test.ts

```diff
@@ -0,0 +1,208 @@
+import { dataOffset, debugId, refNumOffset, sizeOffset, typeOffset } from "../memory";
+import { assertPrint, assertTCFail, assertTC, assertFail, assertMemState } from "./asserts.test";
+import { NUM, BOOL, NONE, CLASS } from "./helpers.test"
+
+describe("Memory tests", () => {
+
+assertMemState("classes-from-object", `
+    class Rat(object):
+        id: int = 123
+        y: int = 0
+        def __init__(self: Rat):
+            self.y = 1
+
+    x: Rat = None
+    x = Rat()
+  `, [
+    // first value in the tuple denotes id, NOTE: this is a hack since we dont have access to object names
```

Comment:
I'm a little confused about this – if you have heap access isn't it deterministic where the first-created object will go on the heap, so you can just check heap contents?

Reply:
When we do start reclaiming memory and move stuff around in the heap, I thought this would be easier to do. We could totally switch to calculating addresses though, but I feel it just makes writing more complex tests a bit harder. For e.g if we move stuff around in the heap, we would have to know where it got moved to so that we can check the values at that address